easybuild/easyblocks/s/scipion.py

```diff
@@ -0,0 +1,165 @@
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing Scipion, implemented as an easyblock
+
+@author: Ake Sandgren (HPC2N, Umea University)
+"""
+import os
+
+import easybuild.tools.environment as env
+from easybuild.easyblocks.generic.scons import SCons
+from easybuild.tools.filetools import apply_regex_substitutions, change_dir, mkdir, symlink
+from easybuild.tools.modules import get_software_root
+from easybuild.tools.run import run_cmd
+
+
+class EB_Scipion(SCons):
+    """Support for building/installing Scipion."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize Scipion-specific variables."""
+        super(EB_Scipion, self).__init__(*args, **kwargs)
+
+    def configure_step(self):
+        """Custom configuration procedure for Scipion."""
+        # Setup scipion.conf, protocols.conf, and host.conf
+        # scipion.conf needs to be configured with paths to all the
+        # dependencies, stupid but that's the way it works at the
+        # moment.
+
+        # Create the config files and then make the subtitutions
+        cmd = './scipion --config %s/config/scipion.conf config' % self.installdir
+        run_cmd(cmd, log_all=True, simple=True)
+
+        # Mapping of EB software names and variables to patch in
+        # scipion.conf
+        map_sw2var_name = {
+            'ctffind': 'CTFFIND4_HOME',
+            'EMAN2': 'EMAN2DIR',
+            'frealign': 'FREALIGN_HOME',
+            'MotionCor2': 'MOTIONCOR2_HOME',
+            'RELION': 'RELION_HOME',
+            'Java': 'JAVA_HOME',
+            'VMD': 'VMD_HOME',
+            'MATLAB': 'MATLAB_DIR',
+        }
+
+        # Some other variables we need to change
+        regex_subs = [
+            (r'^CC = .*$', r'CC = %s' % os.getenv('CC')),
+            (r'^CXX = .*$', r'CXX = %s' % os.getenv('CXX')),
+            (r'^LINKERFORPROGRAMS = .*$', r'LINKERFORPROGRAMS = %s' % os.getenv('CXX')),
+            (r'^MPI_CC = .*$', r'MPI_CC = %s' % os.getenv('MPICC')),
+            (r'^MPI_CXX = .*$', r'MPI_CXX = %s' % os.getenv('MPICXX')),
+            (r'^MPI_LINKERFORPROGRAMS = .*$', r'MPI_LINKERFORPROGRAMS = %s' % os.getenv('MPICXX')),
+            (r'^MPI_LIBDIR = .*$', r'MPI_LIBDIR = /'),
+            (r'^MPI_INCLUDE = .*$', r'MPI_INCLUDE = /'),
+            (r'^MPI_BINDIR = .*$', r'MPI_BINDIR = /'),
+            # The EB motioncor2 build renames that binary to an explicit name
+            (r'^MOTIONCOR2_BIN = .*$', r'MOTIONCOR2_BIN = motioncor2'),
+        ]
+        for (k, v) in map_sw2var_name.items():
```

Comment:
please avoid single-letter variables outside of list comprehensions, use `(key, val)` instead